utils/common/src/test/kotlin/ArchiveUtilsTest.kt

```diff
@@ -59,175 +59,39 @@ class ArchiveUtilsTest : WordSpec() {
-                val fileB = outputDir.resolve("dir/b")
-
-                fileA shouldBe aFile()
-                fileA.readText() shouldBe "a\n"
-                fileB shouldBe aFile()
-                fileB.readText() shouldBe "b\n"
-            }
-
-            "unpack with a filter" {
-                val archive = File("src/test/assets/test.tar.gz")
-
-                archive.unpack(outputDir, filter = A_FILTER)
-
-                val fileA = outputDir.resolve("a")
-                val fileB = outputDir.resolve("dir/b")
-
-                fileA shouldBe aFile()
-                fileB shouldNotBe aFile()
-            }
-        }
-
-        "Tar bzip2 archives" should {
-            "unpack" {
-                val archive = File("src/test/assets/test.tar.bz2")
-
-                archive.unpack(outputDir)
-
-                val fileA = outputDir.resolve("a")
-                val fileB = outputDir.resolve("dir/b")
-
-                fileA shouldBe aFile()
-                fileA.readText() shouldBe "a\n"
-                fileB shouldBe aFile()
-                fileB.readText() shouldBe "b\n"
-            }
-
-            "unpack with a filter" {
-                val archive = File("src/test/assets/test.tar.bz2")
-
-                archive.unpack(outputDir, filter = A_FILTER)
-
-                val fileA = outputDir.resolve("a")
-                val fileB = outputDir.resolve("dir/b")
-
-                fileA shouldBe aFile()
-                fileB shouldNotBe aFile()
-            }
-        }
-
-        "Tar xz archives" should {
-            "unpack" {
-                val archive = File("src/test/assets/test.tar.xz")
-
-                archive.unpack(outputDir)
-
-                val fileA = outputDir.resolve("a")
-                val fileB = outputDir.resolve("dir/b")
-
-                fileA shouldBe aFile()
-                fileA.readText() shouldBe "a\n"
-                fileB shouldBe aFile()
-                fileB.readText() shouldBe "b\n"
-            }
-
-            "unpack with a filter" {
-                val archive = File("src/test/assets/test.tar.xz")
-
-                archive.unpack(outputDir, filter = A_FILTER)
-
-                val fileA = outputDir.resolve("a")
-                val fileB = outputDir.resolve("dir/b")
-
-                fileA shouldBe aFile()
-                fileB shouldNotBe aFile()
-            }
-        }
-
-        "Zip archives" should {
-            "unpack" {
-                val archive = File("src/test/assets/test.zip")
-
-                archive.unpack(outputDir)
-
-                val fileA = outputDir.resolve("a")
-                val fileB = outputDir.resolve("dir/b")
-
-                fileA shouldBe aFile()
-                fileA.readText() shouldBe "a\n"
-                fileB shouldBe aFile()
-                fileB.readText() shouldBe "b\n"
-            }
+        (enumValues<ArchiveType>().asList() - ArchiveType.DEB - ArchiveType.NONE).forEach { type ->
```

Comment:
What's the reason for the special handling of DEB archives?

Reply:
See its test implementation. It's different form the other, and thus cannot be covered by the parameterized loop.